sdk/couchbase/tests/bucket_management.rs

```diff
@@ -0,0 +1,287 @@
+use crate::common::consistency_utils::{verify_bucket_created, verify_bucket_deleted};
+use crate::common::features::TestFeatureCode;
+use crate::common::test_config::run_test;
+use crate::common::{generate_string_value, try_until};
+use couchbase::bucket_settings::{
+    BucketSettings, CompressionMode, ConflictResolutionType, EvictionPolicyType, StorageBackend,
+};
+use couchbase::durability_level::DurabilityLevel;
+use std::time::Duration;
+
+mod common;
+
+#[test]
+fn test_create_bucket() {
+    run_test(async |cluster| {
+        if !cluster.supports_feature(&TestFeatureCode::BucketManagement) {
+            return;
+        }
+
+        let manager = cluster.buckets();
+
+        let bucket_name = generate_string_value(10);
+        let settings = BucketSettings::new(&bucket_name, 100).flush_enabled(true);
+
+        manager.create_bucket(settings, None).await.unwrap();
+
+        verify_bucket_created(&manager, &bucket_name).await;
+
+        let bucket = manager.get_bucket(&bucket_name, None).await.unwrap();
+
+        manager.delete_bucket(&bucket_name, None).await.unwrap();
+        verify_bucket_deleted(&manager, &bucket_name).await;
+
+        assert_eq!(bucket.name, bucket_name);
+        assert_eq!(bucket.ram_quota_mb, 100);
+        assert_eq!(bucket.flush_enabled, Some(true));
+    })
+}
+
+#[test]
+fn test_update_bucket() {
+    run_test(async |cluster| {
+        if !cluster.supports_feature(&TestFeatureCode::BucketManagement) {
+            return;
+        }
+
+        let manager = cluster.buckets();
+
+        let bucket_name = generate_string_value(10);
+        let settings = BucketSettings::new(&bucket_name, 100).flush_enabled(true);
+
+        manager.create_bucket(settings.clone(), None).await.unwrap();
+
+        verify_bucket_created(&manager, &bucket_name).await;
+
+        let updated_settings = settings.ram_quota_mb(200);
+        manager.update_bucket(updated_settings, None).await.unwrap();
+
+        try_until(
+            tokio::time::Instant::now() + Duration::from_secs(30),
+            Duration::from_millis(100),
+            "Bucket was not created in time",
+            || async {
+                let bucket = manager.get_bucket(&bucket_name, None).await?;
+
+                if bucket.ram_quota_mb == 200 {
+                    Ok(Some(()))
+                } else {
+                    Ok(None)
+                }
+            },
+        )
+        .await;
+
+        manager.delete_bucket(&bucket_name, None).await.unwrap();
+        verify_bucket_deleted(&manager, &bucket_name).await;
+    })
+}
+
+#[test]
+fn test_delete_bucket() {
```

Comment:
Do we need this since we are deleting and verifying in the earlier tests?

Reply:
I think it's just nice to have, it can be run standalone to reproduce any delete issues.